### PR TITLE
Add contract tests for v2

### PR DESCRIPTION
**Pull request by JulianRoesner**
# Short Description
This updates the test_integration_queue tests:
- make them independent from the erica module. Otherwise there is more stuff that we would need to include in the .dockerignore from tests
- Check that the job was performed

What is still missing is that we would need to set a second ERICA_URL for the integration job in the environment to test for version 2 and add it to the pipenv task

# Feedback
- Does that make sense to you?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

**Comment by JulianRoesner** [comment]
The duplicated lines are because of the tests. We probably could parametrize that.

**Comment by JulianRoesner** [comment]
Yes, for the freischaltcodes we are only testing the failure cases. The freischaltcode can not be requested, activated or revocated for an indefinite number of times. Therefore, we are not able to test something else than failure here.

Or do you see a clever alternative?

**Comment by VictorDelCampo** [comment]
> Yes, for the freischaltcodes we are only testing the failure cases. The freischaltcode can not be requested, activated or revocated for an indefinite number of times. Therefore, we are not able to test something else than failure here.
> 
> Or do you see a clever alternative?

Oh sorry now I see it, for the fsc use cases we are only testing the failure cases, fine! And for the tax number validity and the send tax declaration we have the success cases. Good!

There are only 2 minor changes we should do, I think:

1. For the success result, not only test if the result attribute is in the response but also the specific content of the result depending on the use case right?
- This also means testing if result == null for the failure cases
2. For the failure cases testing if the errorCode and errorMessage !== null
- This also means testing if those attributes == null for the success cases.

Just give me the command and I can take over this ticket too!

**Comment by VictorDelCampo** [comment]
@SannyNguyenHung some summing-up of this PR:
Integratoin tests for v2 (test_integration_queue)
- now independent from erica module.
- the getting-the-job-with-200 tests wait now for the worker to complete the job (4 seconds). For the unlock code use cases we can only test the failure case. For the rest we have both success and failure mix
- tests upgraded, so that we also test if we expect a result (!= None) or if we expect errorCode/errorMessage. In the success-cases we also test the actual content of the result

Github secrets
- old  secret ERICA_TESTING_URL updated to ERICA_TESTING_URL_V1 (same value as before)
- secret ERICA_TESTING_URL_V2 (for the k8s url) added

Pipeline:
- Task added for running integration tests v2
- step added for job contract-tests, so that integratoin tests v2 are also run with the same sh script.
- step updated in job contract-tests, so that we set the new Github secret with the old URL (VM) and we can also pass the new env variable for the task.

IMPORTANT: before this PR is merged we must merge first the PR of the infra repo and manually run the workflow for building the image right?

**Review by SannyNguyenHung — APPROVED** [review]
Looking good for me so far :)

File: .github/workflows/cicd-pipeline.yml [hunk 1]
```diff
@@ -198,7 +198,24 @@ jobs:
...
-          -e ERICA_TESTING_URL='${{ secrets.ERICA_TESTING_URL }}' \
+          -e ERICA_TESTING_URL='${{ secrets.ERICA_TESTING_URL_V1 }}' \
+          -e TEST_INTEGRATION_TASK='test-integration' \
+          ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment erica-contract-tests
+        shell: bash
+
+      - name: Run contract tests v2
+        env:
+          NAMESPACE: erica-staging
+          IMAGE_TAG: 460
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply .github/workflows/cicd-pipeline.yml:209]: Yes! Sanny will merge that PR and I will update the image tag to set it dynamically!
